From b9b79d6d1a522c299d2ae12df396b69321490208 Mon Sep 17 00:00:00 2001
From: "Gary D. Gregory" <ggregory@apache.org>
Date: Tue, 1 Nov 2011 02:34:22 +0000
Subject: [PATCH] [VFS-364] Incorrect lazy initialization of static field
 org.apache.commons.vfs2.util.Messages.resources in
 org.apache.commons.vfs2.util.Messages.findMessage(String)

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/vfs/trunk@1195775 13f79535-47bb-0310-9956-ffa450edef68
---
 .../main/java/org/apache/commons/vfs2/util/Messages.java   | 7 +------
 src/changes/changes.xml                                    | 3 +++
 2 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/core/src/main/java/org/apache/commons/vfs2/util/Messages.java b/core/src/main/java/org/apache/commons/vfs2/util/Messages.java
index c59d2b4800..37821c1430 100644
--- a/core/src/main/java/org/apache/commons/vfs2/util/Messages.java
+++ b/core/src/main/java/org/apache/commons/vfs2/util/Messages.java
@@ -34,7 +34,7 @@ public final class Messages
      */
     private static ConcurrentMap<String, MessageFormat> messages =
         new ConcurrentHashMap<String, MessageFormat>();
-    private static ResourceBundle resources;
+    private static final ResourceBundle resources = new CombinedResources("org.apache.commons.vfs2.Resources");
 
     private Messages()
     {
@@ -101,11 +101,6 @@ private static MessageFormat findMessage(final String code)
             return msg;
         }
 
-        // Locate the message
-        if (resources == null)
-        {
-            resources = new CombinedResources("org.apache.commons.vfs2.Resources");
-        }
         final String msgText = resources.getString(code);
         msg = new MessageFormat(msgText);
         messages.putIfAbsent(code, msg);
diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index 559e7c5967..3fb15bbab9 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -23,6 +23,9 @@
 
   <body>
     <release version="2.1" date="TBD" description="">
+      <action issue="VFS-374" dev="ggregory" type="fix">
+        Incorrect lazy initialization of static field org.apache.commons.vfs2.util.Messages.resources in org.apache.commons.vfs2.util.Messages.findMessage(String)Add FileContent write APIs.
+      </action>
       <action issue="VFS-373" dev="ggregory" type="add">
         Add FileContent write APIs.
       </action>
